### ISSUE TITLE
Sonoff local devices not working properly after 2023.4 HASS update

### ISSUE THREAD

**Issue by vgarcive** [AlexxIT/SonoffLAN#1164]
I have 3 sonoff MINI working on local. I suddently lost the ability to turn them on/on since 2023.4.3 update, it is very strange because I can see some entities, although only 9 out of 18 I had before,  like RSSI and they still have IP assigned so I believe they are connected but turning on and off appears always as Unknown and I can´t do anything. (See the picture attached)

I reuploaded the integration several times and nothing happens, I tried to remove and reconfigure the integration and no change. I cannot find any related issue on the internet. Prior to this problem, the integration became unavailable several times but I could fix it reloading the integration, but now I cant control the devices.

![image](https://user-images.githubusercontent.com/124352152/232073626-ef72f10a-1d6d-442d-94a2-4d9f376ea350.png)
![image](https://user-images.githubusercontent.com/124352152/232072288-437a35e6-af03-416a-8e6c-0f8d81fb55c5.png)

I share the logs as well to see if you could help me.
[sonoff log.txt](https://github.com/AlexxIT/SonoffLAN/files/11233843/sonoff.log.txt)

**Comment by Moldiz** [comment]
I've got the exact same issue but only with sonoff mini in dyi mode, D1 in dyi mode seems to be ok. Additionally i have all of them configured in configuration.yaml as lights:
`sonoff:
  devices:
    1000fc9c64: # Lampka nocna sypialnia
      device_class: light
    1000f95586: # Lampa sypialnia
      device_class: light
    1000fa49ac: # Lampa pokój
      device_class: light`

**Comment by lucafe82** [comment]
Same here!

**Comment by Cfoliaco** [comment]
It seems to be something related to cloud and local setup. When configured cloud nothing works but in auto or local some devices works. LED strip only works in cloud.

**Comment by vgarcive** [comment]
yeah the devices I have on cloud working with the Ewelink account, work correctly. But the MINI DIY that I have in local dont work, they seem to be somehow connected to the LAN but do not report all entities.

**Comment by sipimokus** [comment]
@protomaxita see this, maybe help for you: https://github.com/AlexxIT/SonoffLAN/issues/1159#issuecomment-1500877648

**Comment by vgarcive** [comment]
> For me pow R2 not report data for hours. Revert to v. 3.4.0 solve trouble

How do you revert it?

**Comment by gabrymed** [comment]
> > For me pow R2 not report data for hours. Revert to v. 3.4.0 solve trouble
> 
> How do you revert it?

![Immagine](https://user-images.githubusercontent.com/57106603/232859277-00d8d925-5a59-4ea4-96d3-3d0618f37971.jpg)
Click menu up-right then click "download again" and select a previous version

**Comment by gabrymed** [comment]
My Pow R2 doesn't report data in LAN mode with latest version but device works well in ewelink app. I resolve downgrade to 3.3.0 version.

**Comment by vgarcive** [comment]
Yeah it works when downgrading to 3.3.0. Thanks!

**Comment by caco3** [comment]
I can confirm that there seems to be a regression. i changed back to 3.3 and it works again.

Note that reloading alone did not work, I also had to restart HA.

**Comment by brentcohen** [comment]
Downgrading to 3.4 seems to have resolved the problem for my Micro USB switches when blocked from cloud access.

**Comment by AlexxIT** [comment]
DIY devices was fixed at the beginning of April https://github.com/AlexxIT/SonoffLAN/commit/59c4066c70a5ce42de69df6fdabf8f9c31fd4305

**Comment by espilioto** [comment]
Can confirm; MINI R2 broken in 3.5 and 3.5.1, working with 3.4 :(

**Comment by admfuby** [comment]
Hi,
I have the same problem with the last version of this integration.
My HACS only permits to download the version 3.5.4 as the oldest version.
Any idea to download a specific version from HACS?
![image](https://github.com/AlexxIT/SonoffLAN/assets/168348397/a2b70aaf-1331-46c6-bc3b-77c823962dc9)
Thx all!

**Comment by sipimokus** [comment]
Hello @admfuby!

I think that HACS show maximum 5 of the releases, you can install manually older version, or maybe you can "edit / workaround" your HACS.
https://github.com/hacs/integration/blob/main/custom_components/hacs/utils/configuration_schema.py#L46

**Comment by admfuby** [comment]
Hi @sipimokus!
Thx for your contribution! I can configure the maximum of the releases in the integration menu of HACS.
